src/lib/ngsi/ContextElement.cpp

```diff
@@ -111,6 +111,32 @@ std::string ContextElement::render(ConnectionInfo* ciP, RequestType requestType,
 
 /* ****************************************************************************
 *
+* ContextElement::toJson - 
+*/
+std::string ContextElement::toJson(ConnectionInfo* ciP, NotificationFormat notifyFormat)
```

Comment:
Don't miss "symmetry" with "exception". I mean, if some toJson() needs ciP, let's add it. But for the most of them that doesn't need, no need to make things more complex. 

I love the `toJson(void)` we have in some places and I'd prefer if all toJson() tend to work as closer as possible to that pattern. In my opinion, this is a HUGE improvement compared with the old render() family used mainly in NGSIv1, using very complex signatures that makes the code very difficult to understand and mantain (I had nightmares in the past with all that stuff about isVector, commaBlabla, isLastElement and the like :S)